src/bigBed.c

```diff
@@ -0,0 +1,33 @@
+#include "ucsc/common.h"
+#include "ucsc/linefile.h"
+#include "ucsc/localmem.h"
+#include "ucsc/bbiFile.h"
+#include "ucsc/bigBed.h"
+
+#include "bigBed.h"
+#include "handlers.h"
+
+/* --- .Call ENTRY POINT --- */
+SEXP BBDFile_seqlengths(SEXP r_filename) {
```

Comment:
Can this share code with the seqlengths accessor for BigWig?

Reply:
Yes it can, but not directly as `bigWigFileOpen` checks for the BigWig file signature so to make it work with BigBed files we need to add a parameter to the `BWGFile_seqlengths` to control on which file it is going to operate on(bigWig or BigBed)
and then it pick right function accordingly(`bigWigFileOpen` or `bigBedFileOpen`)

Reply:
What about extracting the common code into a helper function?

Reply:
Yes, this sounds good!
I will create a helper header file that will contain common code between BigWig and BigBed.